development/extensions/index.md

```diff
@@ -24,51 +24,125 @@ One of the primary aims of BlueOS is to be a platform that is readily extendable
+
+#### Where Does the Code Run?
+
+To integrate well within BlueOS, extensions typically include a [backend service](#backend-services) that runs on the onboard computer (in the vehicle), which provides some functionality, and is often accessed by a [frontend interface](#frontend-interfaces) that runs in the control station computer's browser.
+
+{{ easy_image(src="frontend-backend", width=600, center=true) }}
+
+A common usage process would follow:
+1. Vehicle turns on, and backend service initialises
+2. A user requests the frontend interface, from the browser in their control station computer
+3. Backend serves a web interface to the browser
+4. User interacts with the interface, via some kind of input
+    - button clicks, text submissions, joystick events, etc.
+5. Frontend makes API calls to its backend service to facilitate the desired interaction(s)
+6. Backend receives an API call, processes it, and responds appropriately
+    - processing typically involves logging the request, accessing some hardware, calculating something, and/or requesting something from another service
+7. Frontend receives the response from the backend, and updates the interface accordingly
+    - the frontend may also do some processing, such as decoding an image to display to the user
+8. Repeat from step 4. until further interaction is no longer required
+
+##### Backend Services
+- Communicate with each other (via [APIs](../../advanced-usage/#available-services))
+- Typically run continuously while the vehicle is on
+- Have access to the vehicle computer hardware, including connected devices and the filesystem
+- May have internet access, if the vehicle computer does
+- May serve a graphical web-interface that's accessible via the frontend
+    - Extensions without a graphical interface could include device drivers and/or something with fully pre-determined functionality (that doesn't require user input)
+
+##### Frontend Interfaces
+- Only run when opened by a user
+    - Could be run by multiple users simultaneously, on separate devices that are on a shared network
+- Can do some processing
+    - This can be useful to reduce the amount of data that needs to be sent from the vehicle to the control station computer (e.g. send an encoded image, which the frontend decodes and displays), or to offload some processing/calculations that would otherwise need to be done by the onboard computer
+- Can communicate with their backend (via its API)
+    - It is also possible to communicate directly with other service APIs, but then the backend doesn't know about that interaction, which can make logging and debugging more challenging
```

Comment:
- Can work as a standalone application
    - You can add all the necessary functionality in the frontend and no backend would be needed, only an HTTP server to provide the frontend files for the static webpage. You could create calculators, image processing tools, and etc.